nebula-logger/core/tests/log-management/classes/LoggerSObjectHandler_Tests.cls

```diff
@@ -154,46 +154,57 @@ private class LoggerSObjectHandler_Tests {
+            SObjectHandlerApexClass__c = 'Some_Fake_Apex_Class'
+        );
+        LoggerPlugin.setMock(plugin);
         MockLogHandler handler = new MockLogHandler();
         handler.triggerOperationType = TriggerOperation.BEFORE_INSERT;
         handler.triggerNew = new List<Log__c>{ new Log__c(TransactionId__c = '1234') };
         handler.triggerNewMap = null;
         handler.triggerOldMap = null;
-        LoggerPlugin__mdt plugin = new LoggerPlugin__mdt(
-            IsEnabled__c = true,
-            IsLogEnabled__c = true,
-            PluginApiName__c = 'Some_Fake_Apex_Class',
-            PluginType__c = 'SObject Handler - Apex'
-        );
-        handler.addMockPlugin(plugin);
 
         handler.execute();
 
         System.assertEquals(1, handler.executionCount);
-        System.assertEquals(1, handler.getPluginConfigurations().size());
-        System.assertEquals(0, handler.getExecutedApexPlugins().size());
+        System.assertEquals(1, handler.getPluginConfigurations().size(), handler.getPluginConfigurations());
+        System.assertEquals(0, handler.getExecutedApexPlugins().size(), handler.getExecutedApexPlugins());
     }
 
     @IsTest
     static void it_should_gracefully_skip_non_existent_flow_plugin() {
+        LoggerPlugin__mdt plugin = new LoggerPlugin__mdt(DeveloperName = 'Mock_Plugin', IsEnabled__c = true, SObjectHandlerFlowName__c = 'Some_Fake_Flow');
+        LoggerPlugin.setMock(plugin);
         MockLogHandler handler = new MockLogHandler();
         handler.triggerOperationType = TriggerOperation.BEFORE_INSERT;
         handler.triggerNew = new List<Log__c>{ new Log__c(TransactionId__c = '1234') };
         handler.triggerNewMap = null;
         handler.triggerOldMap = null;
+
+        handler.execute();
+
+        System.assertEquals(1, handler.executionCount);
+        System.assertEquals(1, handler.getPluginConfigurations().size(), handler.getPluginConfigurations());
+        System.assertEquals(0, handler.getExecutedApexPlugins().size(), handler.getExecutedApexPlugins());
+    }
+
+    @IsTest
+    static void it_should_execute_apex_plugin() {
         LoggerPlugin__mdt plugin = new LoggerPlugin__mdt(
+            DeveloperName = 'ExampleApexPlugin',
             IsEnabled__c = true,
-            IsLogEnabled__c = true,
-            PluginApiName__c = 'Some_Fake_Flow',
-            PluginType__c = 'SObject Handler - Flow'
+            SObjectHandlerApexClass__c = MockLogHandlerPlugin.class.getName()
         );
-        handler.addMockPlugin(plugin);
+        LoggerPlugin.setMock(plugin);
 
-        handler.execute();
+        Log__c log = new Log__c(LoggedBy__c = UserInfo.getUserId(), TransactionId__c = '6789');
+        insert log;
 
-        System.assertEquals(1, handler.executionCount);
-        System.assertEquals(1, handler.getPluginConfigurations().size());
-        System.assertEquals(0, handler.getExecutedApexPlugins().size());
+        String expectedProfileName = 'Some String';
```

Comment:
It really is a shame that inner classes can't have static inner properties - how do you feel about promoting this to top-level constant in this class?

Reply:
Ooooh, yeah, I love the idea of promoting it to a top-level constant! I'll make that change. It's definitely a shame that inner classes can't have static stuff, I run into frustrations with that far too often.